runtime/j9vm/jvm.c

```diff
@@ -4234,95 +4579,208 @@ JVM_NativePath(char* path)
+#if !defined(J9ZTPF)
+			newSignalAction.sa_flags = SA_RESTART;
+#else /* !defined(J9ZTPF) */
+			newSignalAction.sa_flags = 0;
+#endif /* !defined(J9ZTPF) */
+			if ((void *)J9_PRE_DEFINED_HANDLER_CHECK == handler) {
+				newSignalAction.sa_handler = (void (*)(int))dummySignalHandler;
+			} else {
+				newSignalAction.sa_handler = (void (*)(int))handler;
+			}
+			OMRSIG_SIGACTION(sigNum, &newSignalAction, &oldSignalAction);
+#endif /* defined(WIN32) */
+		}
+	}
+
+#if defined(WIN32)
+	if (((void *)J9_SIG_ERR != oldHandler) && (handler == oldHandler)) {
+		oldHandler = (void *)J9_OLDHANDLER_SAME_AS_NEWHANDLER;
+	} else {
+		oldHandler = (void *)oldHandler;
+	}
+#else /* defined(WIN32) */
+	if ((NULL != oldSignalAction.sa_handler)
+			&& (newSignalAction.sa_handler == oldSignalAction.sa_handler)) {
+		oldHandler = (void *)J9_OLDHANDLER_SAME_AS_NEWHANDLER;
+	} else {
+		oldHandler = (void *)oldSignalAction.sa_handler;
+	}
+#endif /* defined(WIN32) */
+
+	return oldHandler;
 }
 
 
 /**
- * Method stub for method not yet implemented.
+ *  Return the integer value of the signal given the name of the
+ *  signal.
+ *
+ *  @param  sigName Name of the signal
+ *
+ *  @returns Integer value of the signal on success
+ *           J9_SIG_ERR (-1) on failure
+ *
+ *	DLL: jvm
  */
 jint JNICALL
-JVM_FindSignal(const char* sigName)
+JVM_FindSignal(const char *sigName)
 {
+	const J9SignalMapping *mapping = NULL;
+	jint signalValue = J9_SIG_ERR;
+	BOOLEAN nameHasSigPrefix = FALSE;
+	const char *fullSigName = sigName;
+#if !defined(WIN32)
+	char nameWithSIGPrefix[J9_SIGNAME_BUFFER_LENGTH] = {0};
+#endif /* !defined(WIN32) */
 
 	Trc_SC_FindSignal_Entry(sigName);
 
-#define J9_MAP_SIGNAL(ID) ID##U
-#define J9_RETURN_SIGNAL(ID) \
-	const int ID##U = -1; Trc_SC_FindSignal_Exit(J9_MAP_SIGNAL(ID)); return J9_MAP_SIGNAL(ID);
-#define J9_MATCH_SIGNAL(ID) \
-	if (strcmp(sigName, #ID) == 0) {J9_RETURN_SIGNAL(SIG##ID)}
-
-	/* POSIX */
-	J9_MATCH_SIGNAL(HUP)
-	J9_MATCH_SIGNAL(INT)
-	J9_MATCH_SIGNAL(QUIT)
-	J9_MATCH_SIGNAL(ABRT)
-	J9_MATCH_SIGNAL(KILL)
-	J9_MATCH_SIGNAL(ALRM)
-	J9_MATCH_SIGNAL(TERM)
-
-	/* POSIX 1003.1-2001 */
-	J9_MATCH_SIGNAL(BUS)
-	J9_MATCH_SIGNAL(CHLD)
-	J9_MATCH_SIGNAL(CONT)
-	J9_MATCH_SIGNAL(FPE)
-	J9_MATCH_SIGNAL(ILL)
-	J9_MATCH_SIGNAL(PIPE)
-	J9_MATCH_SIGNAL(SEGV)
-	J9_MATCH_SIGNAL(STOP)
-	J9_MATCH_SIGNAL(TSTP)
-	J9_MATCH_SIGNAL(TTIN)
-	J9_MATCH_SIGNAL(TTOU)
-	J9_MATCH_SIGNAL(URG)
-	J9_MATCH_SIGNAL(USR1)
-	J9_MATCH_SIGNAL(USR2)
-
-	/* POSIX 1003.1-2001 with XSI */
-	J9_MATCH_SIGNAL(POLL)
-	J9_MATCH_SIGNAL(PROF)
-	J9_MATCH_SIGNAL(SYS)
-	J9_MATCH_SIGNAL(TRAP)
-	J9_MATCH_SIGNAL(VTALRM)
-	J9_MATCH_SIGNAL(XCPU)
-	J9_MATCH_SIGNAL(XFSZ)
-
-	/* Miscellaneous */
-	J9_MATCH_SIGNAL(IO)
-	J9_MATCH_SIGNAL(WINCH)
-	J9_MATCH_SIGNAL(PWR)
-	J9_MATCH_SIGNAL(UNUSED)
-	J9_MATCH_SIGNAL(BREAK)
-
-	Trc_SC_FindSignal_Exit(-1);
-	return -1;
+	if (NULL != fullSigName) {
+		mapping = signalMap;
+		size_t sigPrefixLength = strlen(J9_SIG_PREFIX);
```

Comment:
This isn't valid C code: declarations must be at beginning of the block.
(the assignment could move into the initializer of a `for` loop replacing the `while` loop below.)

Reply:
fixed.